src/utilities/TypeInfo.ts

```diff
@@ -38,6 +39,14 @@ import type { GraphQLSchema } from '../type/schema.js';
 import { typeFromAST } from './typeFromAST.js';
 import { valueFromAST } from './valueFromAST.js';
 
+export interface GraphQLSignature {
```

Comment:
Alternatively, we could introduce a new `GraphQLFragmentVariableSignature` interface. If we wanted we could also have a union `type GraphQLSignature =  GraphQLArgument | GraphQLFragmentVariableSignature`.

I believe the basic learning from this PR is that we can't have a union over `VariableDefinitionNode` itself, because we have to validate the type and generate the actual default value.